src/main.cpp

```diff
@@ -12,27 +14,66 @@ json toJson(const ProcessStats &stats) {
   j["max_memory"] = stats.max_memory;
   j["system_time"] = stats.system_time;
   j["user_time"] = stats.user_time;
+  j["timeouted"] = stats.timeouted;
   return j;
 }
 
-static std::vector<std::string> parsArgs(int argc, char **argv) {
+static std::vector<std::string> convertArgs(int argc, char **argv) {
   std::vector<std::string> args(argc - 1);
   std::transform(argv + 1, argv + argc, args.begin(), [](auto s) { return s; });
   return args;
 }
 
-static bool handleShowVersion(std::vector<std::string> args) {
-  if(args.size() == 0 || args.front() != "--version" ) return false;
-  std::cout << "Zmora-runner version " << VERSION << std::endl;
-  return true;
+static bool handleShowVersion(const Config& config) {
+  if(config.showVersion)
+    std::cout << "Zmora-runner version " << VERSION << std::endl;
+  return config.showVersion;
+}
+
+static Config parseArgs(std::vector<std::string>&& vector) {
+  Config config{};
+
+  auto it = begin(vector);
+  for(; it != end(vector); ++it){
+    if(it->find("--") != 0) break;
+
+    if(*it == "--timeout") {
+      try {
+        const auto timeout = stoul(*(++it));
```

Comment:
`*(++it)` będzie poza zakresem, jeżeli w wektorze wejściowym będzie jedynie element `--timeout`.